sklearn/neural_network/tests/test_mlp.py

```diff
@@ -267,6 +268,37 @@ def test_lbfgs_regression():
             assert_greater(mlp.score(X, y), 0.95)
 
 
+@pytest.mark.parametrize('X,y', classification_datasets)
+def test_lbfgs_classification_maxfun(X, y):
+    # Test lbfgs parameter max_fun.
+    # It should independently limit the number of iterations for lbfgs.
+    max_fun = 10
+    # classification tests
+    for activation in ACTIVATION_TYPES:
+        mlp = MLPClassifier(solver='lbfgs', hidden_layer_sizes=50,
+                            max_iter=150, max_fun=max_fun, shuffle=True,
+                            random_state=1, activation=activation)
```

Comment:
Doesn't this raise a ConvergenceWarning? If so we should assert it,
```
with pytest.warns(ConvergenceWarning):
   mlp.fit(...)
```
If it doesn't, I think it might make sense to add it.

Reply:
Good catch again.  Yes it doesn't look like we were raising convergence warnings when LBFGS did not converge.  I've added that, as well as the line you suggested to the tests.  The tests now still pass when I run them locally.
Thank you for the very useful feedback.